### DISCUSSION TITLE
change the options api

### DISCUSSION THREAD

**Discussion by StockMarketDev** [guanquann/Stocksera#39] [Ideas]
I think it would be a good idea to change from using TD as an api to get stock options data as they are not allowing new users into their developer programme. overall amazing project

**Comment by guanquann** [comment]
Hi. Do you have an alternative data source? TD is not accepting any more signups for their API because they are migrating to a new API in the near future. Options data from Yahoo Finance seems very unstable when I tried it in the past

**Comment by StockMarketDev** [comment]
Finnhub API: This API provides real-time options data for US stocks and ETFs, including strike prices, expiration dates, and more.
Polygon.io API: This API provides access to real-time options data for US markets, including strike prices, expiration dates, and more.
Barchart API: This API offers access to real-time options data for US stocks and ETFs, including strike prices, expiration dates, and more.
Intrinio API: This API provides access to real-time and historical options data for various markets, including US and international markets.
Tradier API: This API provides real-time options data for various stocks, including strike prices and expiration date

Also thers another problem we cant seem to be able to solve the fact that we contiously keep getting 503 error and in order to get it ack and running we must execcute the docker compose up -d , any idea how we could fix this

**Reply by guanquann** [reply]
> Also thers another problem we cant seem to be able to solve the fact that we contiously keep getting 503 error and in order to get it ack and running we must execcute the docker compose up -d , any idea how we could fix this

Error 503 means that the server is currently unable to handle the request due to a temporary overloading of the server.

**Reply by StockMarketDev** [reply]
we got a problem with the script where , all data is not updating at the times given unless we run the command docker compose run --rm web python tasks_to_run.py nothing will update, any tips ?

**Reply by guanquann** [reply]
`python tasks_to_run.py` is customisable as you can change the boolean value of the variable. The one I have now is a simple template to run all the tasks at once. You will need to change the settings of the file according to your preference. Hope this helps!
